File: app/src/main/java/com/tutv/android/db/dao/UserDao.java
package com.tutv.android.db.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.tutv.android.domain.User;

@Dao
public interface UserDao {
    @Query("SELECT * FROM user LIMIT 1")
    User getCurrentUser();

    @Insert
    void insert(User user);

    @Delete
    void delete(User user);
}

File: app/src/main/java/com/tutv/android/db/dao/UserDao.kt
package com.tutv.android.db.dao

import androidx.room.*
import com.tutv.android.domain.*

@Dao
interface UserDao {
    @get:Query("SELECT * FROM user LIMIT 1")
    val currentUser: User?

    @Insert
    fun insert(user: User)

    @Delete
    fun delete(user: User)
}